CRM/MembershipExtras/Hook/Pre/Membership.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+/**
+ * Implements hook to be run before a membership is created/edited.
+ */
+class CRM_MembershipExtras_Hook_Pre_Membership {
+
+  /**
+   * Parameters that will be used to create the membership.
+   *
+   * @var array
+   */
+  private $params;
+
+  /**
+   * Action being performed on the membership.
+   *
+   * @var string
+   */
+  private $op;
+
+  /**
+   * ID of the membership.
+   *
+   * @var int
+   */
+  private $id;
+
+  /**
+   * ID of the contribution that was created before the action was called for
+   * the membership.
+   *
+   * @var int
+   */
+  private $contributionID;
+
+  public function __construct($op, $id, &$params, $contributionID) {
+    $this->op = $op;
+    $this->id = $id;
+    $this->params = &$params;
+    $this->contributionID = $contributionID;
+  }
+
+  /**
+   * Preprocesses parameters used for Membership operations.
+   */
+  public function preProcess() {
+    switch ($this->op) {
+      case 'create':
+        $this->preProcessCreation();
+        break;
+
+      case 'edit':
+        $this->preProcessEdit();
+        break;
+    }
+  }
+
+  /**
+   * Fixes parameters used to create a membership.
+   */
+  private function preProcessCreation() {

```

Comment:
again, these calculations will probably screw the price-set payments, it should only be applied to financial type payments.

Reply:
Same, tested and all works ok.